zingolib/src/wallet/traits.rs

```diff
@@ -991,21 +949,29 @@ where
         note_and_metadata: &D::WalletNote,
         spend_key: Option<&D::SpendingKey>,
     ) -> Option<Self> {
-        // Include only non-0 value notes that haven't been spent, or haven't been included in an unconfirmed spend yet.
+        // Include only non-0 value notes that haven't been spent, or haven't been included
+        // in an unconfirmed spend yet.
         if note_and_metadata.spent().is_none()
             && note_and_metadata.pending_spent().is_none()
             && spend_key.is_some()
-            && !note_and_metadata.pending_receipt()
             && note_and_metadata.value() != 0
         {
-            Some(Self::from_parts_unchecked(
-                transaction_id,
+            // Filter out notes with nullifier or position not yet known
```

Comment:
This seems like it should be its own `fn` with a doc-comment, and perhaps a doc-test.

Reply:
The if clause, you mean? Sure...can do